okhttp/src/main/java/okhttp3/Headers.java

```diff
@@ -334,6 +339,17 @@ public Builder addAll(Headers headers) {
       return this;
     }
 
+    /**
+     * Add a Date header with the current time based on
+     * https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date .
```

Comment:
Do we do this bit for server responses already?

>    A recipient with a clock that receives a response message without a
>    Date header field MUST record the time it was received and append a
>    corresponding Date header field to the message's header section if it
>    is cached or forwarded downstream.

Reply:
I'm not sure. Feels like this would also be done via an interceptor though.